astropy/io/ascii/cds.py

```diff
@@ -164,14 +203,493 @@ def get_cols(self, lines):
                     raise ValueError(f'Line "{line}" not parsable as CDS header')
 
         self.names = [x.name for x in cols]
-
         self.cols = cols
 
+    def _split_float_format(self, value):
+        """
+        Splits a Float string into different parts to find number
+        of digits after decimal and check if the value is in Scientific
+        notation.
+
+        Parameters
+        ----------
+        value : the float value
+
+        Returns
+        -------
+        fmt: (int, int, int, bool, bool)
+            List of values describing the Float sting.
+            (size, dec, ent, sign, exp)
+            size, length of the given string.
+            ent, number of digits before decimal point.
+            dec, number of digits after decimal point.
+            sign, whether or not given value signed.
+            exp, is value in Scientific notation?
+        """
+        regfloat = re.compile(r"""(?P<sign> [+-]*)
+                                  (?P<ent> [^eE.]+)
+                                  (?P<deciPt> [.]*)
+                                  (?P<decimals> [0-9]*)
+                                  (?P<exp> [eE]*-*)[0-9]*""",
+                              re.VERBOSE)
```

Comment:
Though I guess to be fair this expression might work for parsing the bits of a string that is known to have been generated by "normal" Python formatting of a float value. But do remember that the format specifier for a column can be quite general, e.g. `col['a'].info.format = '++{:.2f}'`. The current expression would accept values formatted that way, whereas the right behavior would probably be an exception since a float value is being formatted in a non-float output.

So you might test for a correctly formatted float by doing `float(value)` in a try/except.

Reply:
This would need some work. So, needs an issue.